test/serve/build_delay_test.dart

```diff
@@ -72,52 +54,107 @@ main() {
     await endPubServe();
   });
 
-  test("continual fast edits won't cause multiple builds", () async {
-    var pubServeProcess =
-        await pubServe(forcePoll: false, args: ['--build-delay', '100']);
-    expect(pubServeProcess.stdout,
-        emitsThrough(contains('Build completed successfully')));
-    await requestShouldSucceed("packages/myapp/lib.dart", "foo() => 'foo';");
-
-    for (var i = 0; i < 10; i++) {
-      writeTextFile(libFilePath, "foo() => '$i';");
-      await new Future.delayed(new Duration(milliseconds: 50));
+  group('unit tests', () {
+    Barback barback;
+    List<BuildResult> barbackResults;
+    AssetEnvironment environment;
+    final libAssetId = new AssetId('myapp', 'lib/lib.dart');
+    _MockDirectoryWatcher libWatcher;
```

Comment:
Its the watcher for the `lib` directory (each dir is watched separately),  I added a comment.